src/AppBundle/Service/File/Storage/S3Storage.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace AppBundle\Service\File\Storage;
+
+use Aws\S3\Exception\S3Exception;
+use \Aws\S3\S3Client;
+
+/**
+ * Class to upload/download/delete files from S3
+ *
+ * Original logic
+ * https://github.com/ministryofjustice/opg-av-test/blob/master/public/index.php
+ */
+class S3Storage implements StorageInterface
+{
+    /**
+     * @var S3Client
+     *
+     * https://github.com/aws/aws-sdk-php
+     * http://docs.aws.amazon.com/aws-sdk-php/v2/api/class-Aws.S3.S3Client.html
+     *
+     *
+     * for fake s3:
+     * https://github.com/jubos/fake-s3
+     * https://github.com/jubos/fake-s3/wiki/Supported-Clients
+     */
+    private $s3Client;
+
+    /**
+     * @var string
+     */
+    private $bucketName;
+
+
+    public function __construct(S3Client $s3Client, $bucketName)
```

Comment:
Could we have a docblock on here please? When coding the IDE can then show the parameters expected